src/quacc/utils/coords.py

```diff
@@ -0,0 +1,26 @@
+"""Utilities for working with atomic coordinates."""
+
+from __future__ import annotations
+
+import copy
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from ase import Atoms
+
+
+def perturb(mol: Atoms, vector: list[list[float]], scale: float):
+    mol_copy = copy.deepcopy(mol)
+    mode_copy = copy.deepcopy(vector)
+
+    orig_pos = mol_copy.get_positions()
+
+    if not isinstance(mode_copy, np.ndarray):
+        mode_copy = np.asarray(mode_copy)
```

Comment:
I don't think you modify `vector` in-place so there is no need to copy it.

I'd suggest just doing

```python
mode = np.asarray(vector)
```

and then you can skip both the `deepcopy()` call and the `isinstance` check (i..e just call the `np.asarray`). Note that `np.asarray` doesn't modify in-place, so you're good.